### PR TITLE
Add size property to Picklist component

### PR DESCRIPTION
**Pull request by quadreex**

### CODE REVIEW

**Comment by philippfrank** [comment]
@quadreex I know we have this JIRA ticket number convention in the style guide but I'd say since this is an open source repo and the outside world cannot associate these numbers - we skip them here.

File: src/components/Form/Picklist.js [hunk 1]
```diff
@@ -120,6 +123,8 @@ export class PicklistRaw extends Component {
   }
 }
 
+PicklistRaw.size = PICK_LIST_SIZE;
```
Comment (fspoettel) [inline src/components/Form/Picklist.js]: Why do we need to set this here? There is a lot of indirection going on with the `constants` file and the static property here

File: src/components/Form/constants.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import PropTypes from 'prop-types';
+
+export const PICK_LIST_SIZE = {
+  SMALL: 'small',
```
Comment (fspoettel) [inline src/components/Form/constants.js]: These are pre-defined slds terms, no need to hide them behind constants.

Comment (quadreex) [reply src/components/Form/constants.js]: `PICK_LIST_SIZE` itself is a constant to hide slds internals.

We will use it as:
```
<Picklist
...
size={Picklist.size.SMALL}
```

File: src/components/Form/Picklist.js (hunk 1 above)
Comment (quadreex) [reply src/components/Form/Picklist.js]: We need this static property to hide slds internals from the user of our component:
```
<Picklist
...
size={Picklist.size.SMALL}
>
```

Comment (fspoettel) [reply src/components/Form/Picklist.js]: We do not do this in any other place and this is not necessary as Salesforce does not change these terms and they are also used all over the place in `slds`-classes, which our users (us) have to use all the time. This introduces a new concept that is not documented well and only applied in this component.

Comment (fspoettel) [reply src/components/Form/Picklist.js]: If you think that this should be changed, please open a ticket and we can discuss this as a team. Going forward and changing this in one place while leaving all other components as is is not ideal

**Pull request merged**